packages/esm-patient-allergies-app/src/allergies/allergies-form/allergy-form.test.tsx

```diff
@@ -4,21 +4,30 @@ import { screen, render, within } from '@testing-library/react';
 import { type FetchResponse, showSnackbar } from '@openmrs/esm-framework';
 import { mockAllergens, mockAllergicReactions } from '__mocks__';
 import { mockPatient } from 'tools';
-import { type NewAllergy, saveAllergy, useAllergens, useAllergicReactions } from './allergy-form.resource';
+import {
+  type NewAllergy,
+  saveAllergy,
+  useAllergens,
+  useAllergicReactions,
+  updatePatientAllergy,
+} from './allergy-form.resource';
 import AllergyForm from './allergy-form.component';
-import { AllergenType } from '../../types';
+import { AllergenType, ReactionSeverity } from '../../types';
+import { mockAllergy } from '__mocks__';
 
 const mockSaveAllergy = saveAllergy as jest.Mock<Promise<FetchResponse>>;
 const mockUseAllergens = useAllergens as jest.Mock;
 const mockUseAllergicReactions = useAllergicReactions as jest.Mock;
 const mockShowSnackbar = showSnackbar as jest.Mock;
+const mockUpdatePatientAllergy = updatePatientAllergy as jest.Mock;
 
 jest.mock('./allergy-form.resource', () => {
   const originalModule = jest.requireActual('./allergy-form.resource');
 
   return {
     ...originalModule,
     saveAllergy: jest.fn(() => Promise.resolve({ data: {}, status: 201, statusText: 'Created' })),
+    updatePatientAllergy: jest.fn(() => Promise.resolve({ data: {}, status: 200, statusText: 'Updated' })),
```

Comment:
```suggestion
    saveAllergy: jest.fn().mockResolvedValue({ data: {}, status: 201, statusText: 'Created' });
    updatePatientAllergy: jest.fn().mockResolvedValue({ data: {}, status: 200, statusText: 'Updated' });
```

Nit: You can use the [mockResolvedValue](https://jestjs.io/docs/mock-function-api#mockfnmockresolvedvaluevalue) shorthand to mock a resolved promise.